pkg/ipam/allocator/provider.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package allocator
+
+import (
+	"github.com/cilium/cilium/pkg/ipam"
+)
+
+// AllocatorProvider defines the functions of IPAM provider front-end
+// these are implemented by e.g. pkg/ipam/allocator/{aws,azure}.
+
```

Comment:
Nit: drop this empty line so the godoc comment gets associated with `type AllocatorProvider`.

Reply:
This is still unresolved

Reply:
Are you sure?